app/scripts/modules/appengine/src/serverGroup/configure/wizard/basicSettings.controller.ts

```diff
@@ -104,6 +108,54 @@ class AppengineServerGroupBasicSettingsCtrl implements IController {
     }
   }
 
+  public readonly excludedGcsArtifactTypes = [
+    ArtifactTypePatterns.DOCKER_IMAGE,
+    ArtifactTypePatterns.KUBERNETES,
+    ArtifactTypePatterns.FRONT50_PIPELINE_TEMPLATE,
+    ArtifactTypePatterns.EMBEDDED_BASE64,
+    ArtifactTypePatterns.MAVEN_FILE,
+    ArtifactTypePatterns.JENKINS_FILE,
+    ArtifactTypePatterns.HTTP_FILE,
+    ArtifactTypePatterns.GITHUB_FILE,
+  ];
+
+  public readonly excludedContainerArtifactTypes = [
+    ArtifactTypePatterns.KUBERNETES,
+    ArtifactTypePatterns.FRONT50_PIPELINE_TEMPLATE,
+    ArtifactTypePatterns.EMBEDDED_BASE64,
+    ArtifactTypePatterns.MAVEN_FILE,
+    ArtifactTypePatterns.JENKINS_FILE,
+    ArtifactTypePatterns.HTTP_FILE,
+    ArtifactTypePatterns.GITHUB_FILE,
+    ArtifactTypePatterns.GCS_OBJECT,
+  ];
+
+  public onExpectedArtifactEdited = (artifact: IArtifact): void => {
+    this.$scope.$applyAsync(() => {
+      this.$scope.command.expectedArtifactId = null;
+      this.$scope.command.expectedArtifact = artifact;
+    });
+  };
+
+  public onExpectedArtifactSelected = (expectedArtifact: IExpectedArtifact): void => {
+    this.onChangeExpectedArtifactId(expectedArtifact.id);
+  };
+
+  public onChangeExpectedArtifactId = (artifactId: string): void => {
+    this.$scope.$applyAsync(() => {
+      this.$scope.command.expectedArtifactId = artifactId;
+      this.$scope.command.expectedArtifact = null;
+    });
+  };
+
+  public onExpectedArtifactAccountSelected = (accountName: string): void => {
+    this.$scope.$applyAsync(() => {
+      this.$scope.command.storageAccountName = accountName;
+    });
+  };
+
+  public updatePipeline = (_pipeline: IPipeline): void => {};
```

Comment:
Curious why we need this noop?

Reply:
I missed something in my git stash. But It apparently worked as a no-op. I've added the code from my stash back just in case.
https://github.com/spinnaker/deck/pull/7147/commits/5e4de5f960dd5c228ce2d7b2765b73d7e4ca4b26#diff-b17ed7a30567ee09f9205d401ee16cd9R157